### PR TITLE
Add scm-import/export support

### PR DESCRIPTION
**Pull request by scopenco**
This PR  extends the functionality of current `project` resource and add support for scm-import/export settings.

### CODE REVIEW

File: resources/project.rb [hunk 1]
```diff
@@ -123,6 +146,72 @@
...
+      user     'rundeck'
+      group    'rundeck'
+      mode     '0660'
+      cookbook new_resource.cookbook
+      variables(properties: scm_export)
+    end
+  end
+
+  # configure scm_import
+  if new_resource.scm_import
```
Comment (kamaradclimber) [inline resources/project.rb]: This part coyld be factorized with export

File: resources/project.rb [hunk 2]
```diff
@@ -62,6 +73,18 @@
             end
           })
 
+# <> @property scm-import setting of the project
+property :scm_import,
+          kind_of: [FalseClass, Hash],
+          required: false,
+          default: false
```
Comment (kamaradclimber) [inline resources/project.rb]: Dzfault should be nil instead of false

File: resources/project.rb [hunk 3]
```diff
@@ -62,6 +73,18 @@
             end
           })
 
+# <> @property scm-import setting of the project
+property :scm_import,
+          kind_of: [FalseClass, Hash],
```
Comment (kamaradclimber) [inline resources/project.rb]: Remove falseclass supprt

Comment (scopenco) [reply resources/project.rb]: fixed

File: resources/project.rb (hunk 2 above)
Comment (scopenco) [reply resources/project.rb]: fixed

File: resources/project.rb (hunk 1 above)
Comment (scopenco) [reply resources/project.rb]: Can you clarify what do you mean?

Comment (kamaradclimber) [reply resources/project.rb]: lines 200-216 are nearly identical to lines 171-187. I suggest we avoid duplicating this code.
For instance:
```
def scm_config(type)
  scm_config = {}
  scm_config["scm.#{type}.username"] = 'rundeck'
  scm_config["scm.#{type}.config.branch"] = 'master'
  ...
  scm_config
end


template ::File.join(node['rundeck_server']['datadir'], 'projects', new_resource.name, 'etc', 'scm-import.properties') do
       source   'properties.erb'
       user     'rundeck'
       group    'rundeck'
       mode     '0660'
       cookbook new_resource.cookbook
       variables(properties: scm_config('import'))
     end

```

Comment (scopenco) [reply resources/project.rb]: OK, done

**Comment by scopenco** [comment]
@kamaradclimber all done from me

**Comment by kamaradclimber** [comment]
perfeect thanks a lot

**Pull request merged**